dbms/src/Storages/StorageHDFS.cpp

```diff
@@ -138,12 +187,23 @@ BlockInputStreams StorageHDFS::read(
     size_t max_block_size,
     unsigned /*num_streams*/)
 {
-    return {std::make_shared<HDFSBlockInputStream>(
-        uri,
-        format_name,
-        getSampleBlock(),
-        context_,
-        max_block_size)};
+    Strings path_parts;
+    const size_t begin_of_path = uri.find('/', uri.find("//") + 2);
+    const String path_from_uri = uri.substr(begin_of_path);
```

Comment:
Maybe it's possible to use StringRef here? We don't change parts after `substr`, so we can save just pointer and size.

Reply:
Not so obvious....